index.js

```diff
@@ -303,6 +303,29 @@ export const buyPromotedProduct = () => Platform.select({
   android: async() => Promise.resolve(),
 })();
 
+/**
+ * Buy products or subscriptions with offers (iOS only)
+ *
+ * Runs the payment process with some infor you must fetch
+ * from your server.
+ * @param {string} sku The product identifier
+ * @param {string} forUser  An user identifier on you system
+ * @param {object} withOffer The offer information
+ * @param {string} withOffer.identifier The offer identifier
+ * @param {string} withOffer.keyIdentifier Key identifier that it uses to generate the signature
+ * @param {string} withOffer.nonce An UUID returned from the server
+ * @param {string} withOffer.signature The actual signature returned from the server
+ * @param {number} withOffer.timestamp The timestamp of the signature
+ * @returns {Promise}
+ */
+export const buyProductWithOffer = (sku, forUser, withOffer) => Platform.select({
```

Comment:
What about renaming this to `buyProductWithOfferIOS`?

Reply:
Right, I'll do this.

Reply:
Done